grabbit/src/main/groovy/com/twcable/grabbit/spring/batch/repository/GrabbitExecutionContextDao.groovy

```diff
@@ -0,0 +1,24 @@
+package com.twcable.grabbit.spring.batch.repository
+
+import org.springframework.batch.core.repository.dao.ExecutionContextDao
+import org.springframework.batch.item.ExecutionContext;
+
+
+/**
+ * Modified DAO Interface for persisting and retrieving {@link ExecutionContext}
+ * @see ExecutionContextDao for more details
+ */
+interface GrabbitExecutionContextDao extends ExecutionContextDao {
+
+    /**
+     * Returns job execution context paths by comparing "executionId" property on "executionContext/job/<id>" with
+     * "executionId" property on JobExecutions for the @param jobExecutionResourcePaths
+     */
+    public List<String> getJobExecutionContextPaths(List<String> jobExecutionResourcePaths)
```

Comment:
I think having the Dao's do this kind of lifting is a good move

However, I'm wondering if there is a better way to adapt between these DAOs and avoid passing primitives around. I often get lost in our sea of "needy" interfaces.